tests/integration/cattletest/core/common_fixtures.py

```diff
@@ -231,6 +231,24 @@ def wait_transitioning(client, obj, timeout=DEFAULT_TIMEOUT):
     return client.wait_transitioning(obj, timeout=timeout)
 
 
+@pytest.fixture
+def wait_until_expected_state(client, resource, expected_state,
```

Comment:
In the current tests we have the pattern

```python
obj = client.wait_success(obj)
assert obj.state == expected_state
```

Is there a particular advantage to this function over that pattern?